.github/workflows/go.yml

```diff
@@ -1,46 +1,54 @@
-name: Go lint and test
+name: Lint & test
 
 on:
   push:
-    branches: [ master ]
+    branches: [master]
   pull_request:
-    branches: [ master ]
-  workflow_dispatch:   
+    branches: [master]
+  workflow_dispatch:
 
 jobs:
   build:
-    runs-on: self-hosted
+    strategy:
+      matrix:
+        go: [stable]
+        os: [ubuntu-latest, macos-latest, windows-latest]
+    runs-on: ${{ matrix.os }}
     steps:
-    - uses: actions/checkout@v3
-    - name: Set up Go
-      uses: actions/setup-go@v4
-      with:
-        go-version: 1.18
-    - name: Build
-      run: go build -v ./...
+      - uses: actions/checkout@v4
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: 1.22
```

Comment:
Targeting the same version as the minimum [required today in go-ethereum](https://github.com/ethereum/go-ethereum/blob/master/go.mod#L5).